### ISSUE TITLE
imports inside functions are counted as local variables

### ISSUE THREAD

**Issue by PuchatekwSzortach** [PyCQA/pylint#3700]
pylint is counting modules imported inside a function as local variables.
Is this the expected behavior? 

### Steps to reproduce
Run pylint on following, contrived examples:
```python
def my_function():

    # Just a bunch of standard libs to demonstrate the issue
    import os
    import glob
    import json
    import csv
    import functools
    import pickle
    import abc
    import threading
    import multiprocessing
    import importlib
    import base64
    import copy
    import warnings
    import shelve
    import subprocess
    import datetime

    print("nothing much here")

```

### Current behavior
pylint outputs
`sketchpad.py:1:0: R0914: Too many local variables (16/15) (too-many-locals)`

### Expected behavior
I would think module is not consider a variable, and thus not included in locals count.

I appreciate use case I presented is artificial, and error at hand can be side-stepped importing modules outside of functions. I also understand importing modules outside of functions is the correct thing to do, I'm not here to argue this point. But I hope to learn if modules are counted as variables on purpose, or if it's a bug.

### pylint --version output

Tested on two version:
```
pylint 2.5.3
astroid 2.4.2
Python 3.8.3 (default, Jun  9 2020, 17:39:39)
[GCC 8.3.0]
```

and 

```
pylint 1.8.2,
astroid 1.6.1
Python 3.6.8 |Anaconda custom (64-bit)| (default, Dec 29 2018, 19:04:46)
[GCC 4.2.1 Compatible Clang 4.0.1 (tags/RELEASE_401/final)]
```

I appreciate that the design stance might be `everything in python is an object, and module is no exception, thus we count it as a local variable in above case and you are experiencing expected behavior`. I can accept that, but it seems somewhat unintuitive to me and thus decided to raise an issue in case this isn't the expected behavior after all.

**Comment by PCManticore** [comment]
This is expected. Modules are variables as well so they count for this check.
